sstables/mp_row_consumer.hh

```diff
@@ -51,14 +51,36 @@ static inline bytes_view pop_back(std::vector<bytes_view>& vec) {
 class mp_row_consumer_reader : public flat_mutation_reader::impl {
     friend class mp_row_consumer_k_l;
     friend class mp_row_consumer_m;
+protected:
+    shared_sstable _sst;
```

Comment:
We should just fold sstable_mutation_reader and mp_row_consumer_reader (and consumer_* and mp_row_consumer_*). The separation was intended to have a schema-independent sstable reader component with an additional layer adding the schema-aware parts, but we never did and with the m format it's almost impossible.

Not part of this patchset, of course.

Reply:
ACK